honeybee_grasshopper_energy/src/HB Apply Room Schedules.py

```diff
@@ -0,0 +1,187 @@
+            degrees Celcius.
+        lighting_sch_: A fractional for the use of lights over the course of the year.
+            This can also be the name of a schedule to be looked up in the
+            schedule library.
+        electric_equip_sch_: A fractional for the use of electric equipment over
+            the course of the year. This can also be the name of a schedule to
+            be looked up in the schedule library.
+        gas_equip_sch_: A fractional for the use of gas equipment over
+            the course of the year. This can also be the name of a schedule to
+            be looked up in the schedule library.
+        infiltration_sch_: A fractional schedule for the infiltration over the
+            course of the year. This can also be the name of a schedule to
+            be looked up in the schedule library.
+        ventilation_sch_: An optional fractional schedule for the ventilation over the
+            course of the year. This can also be the name of a schedule to be
+            looked up in the schedule library. The fractional values will get
+            multiplied by the total design flow rate (determined from the fields
+            above and the calculation_method) to yield a complete ventilation
+            profile. Setting this schedule to be the occupancy schedule of the
+            zone will mimic demand controlled ventilation.
+    
+    Returns:
+        report: Reports, errors, warnings, etc.
+        rooms: The input Rooms with their schedules edited.
+"""
+
+ghenv.Component.Name = "HB Apply Room Schedules"
+ghenv.Component.NickName = 'ApplyRoomSch'
+ghenv.Component.Message = '0.1.0'
+ghenv.Component.Category = "Energy"
+ghenv.Component.SubCategory = '2 :: Schedules'
+ghenv.Component.AdditionalHelpFromDocStrings = "3"
+
+try:
+    from honeybee_energy.lib.schedules import schedule_by_name
+except ImportError as e:
+    raise ImportError('\nFailed to import honeybee_energy:\n\t{}'.format(e))
+
+try:
+    from ladybug_rhino.grasshopper import all_required_inputs
+except ImportError as e:
+    raise ImportError('\nFailed to import ladybug_rhino:\n\t{}'.format(e))
+
+
+def dup_load(load_obj, input_name, object_name):
+    """Duplicate a Room load object with a check to make sure the object exists."""
+    try:
+        return load_obj.duplicate()
+    except AttributeError:
+        raise ValueError(
+            '{0} input but Room posseses no {1} object.\nUse the the '
+            '"HB Apply Load Values" component to define a {1} '
+            'object.'.format(input_name, object_name))
+
+
+if all_required_inputs(ghenv.Component):
+    # duplicate the initial objects
+    rooms = [obj.duplicate() for obj in _rooms]
+    
+    # assign the occupancy schedule
+    if occupancy_sch_ is not None:
+        if isinstance(occupancy_sch_, str):
+            occupancy_sch_ = schedule_by_name(occupancy_sch_)
+        for room in rooms:
+            people = dup_load(room.properties.energy.people,
```

Comment:
I explained this below.  And what to you mean by "the method itself"? There is a method handling the duplication (that's what dup_load is). Also bear in mind that duplication is only needed because the Grasshopper interface needs to support each component outputting unique objects. So having it called within the methods of the Python libraries is basically forcing the Python libraries to handle the complexity of Grasshopper when, most of the time outside of Grasshopper, this complexity does not exist.

Reply:
> what to you mean by "the method itself"? 

I meant the assignment method which one would use to update a value.